### PR TITLE
Allow using API tokens for Proxmox authentication

### PR DESCRIPTION
**Pull request by mraerino**
Proxmox allows people to create API tokens with limited privileges that can be rotated without deleting a user account. It is usually useful for giving automated systems specific access to parts of the Proxmox API, e.g. a single VM resource pool.

> API tokens allow stateless access to most parts of the REST API by another system, software or API client. Tokens can be generated for individual users and can be given separate permissions and expiration dates to limit the scope and duration of the access. Should the API token get compromised it can be revoked without disabling the user itself.
> _https://pve.proxmox.com/wiki/User_Management#pveum_tokens_

The current authentication flow used for the Proxmox client is not compatible though:
> `POST /api2/json/access/ticket`
> This API endpoint is not available for API tokens.
> _https://pve.proxmox.com/pve-docs/api-viewer/#/access/ticket (POST tab)_

Instead the required auth flow for API tokens is described here:
> To use an API token, set the HTTP header Authorization to the displayed value of the form `PVEAPIToken=USER@REALM!TOKENID=UUID` when making API requests
> _https://pve.proxmox.com/wiki/User_Management#pveum_tokens_

---

This PR introduces a new parameter `token` for the Proxmox builder which allows using the alternative API token flow.
It will take precedence over `password` if both are given.

Tests are provided, docs have been updated.
I tested a development build with a proxmox setup locally.

### CODE REVIEW

**Comment by mraerino** [comment]
Upstream PR is open, let's see which one wins: https://github.com/Telmate/proxmox-api-go/pull/108

**Comment by mraerino** [comment]
now using the method from the client lib

File: builder/proxmox/common/client.go
Comment (carlpett) [inline builder/proxmox/common/client.go:29]: There's a linter failure because these are unused. I suppose they are from before moving the bits upstream?

File: builder/proxmox/common/client.go
Comment (carlpett) [inline builder/proxmox/common/client.go:31]: Nit: This strictly doesn't need to be exported, right?

**Review by carlpett — COMMENTED** [review]
Thanks @mraerino! This all looks really good (getting the token auth into the upstream library was the best way to go IMO), just a linter failure to take care of

**Review by sylviamoss — APPROVED** [review]
Nice! I'll merge it once @carlpett's comments are addressed.

File: builder/proxmox/common/client.go
Comment (mraerino) [reply builder/proxmox/common/client.go]: removed

File: builder/proxmox/common/client.go
Comment (mraerino) [reply builder/proxmox/common/client.go]: fixed

**Comment by mraerino** [comment]
is there something that will let me know once this lands in a stable release?

**Comment by nywilken** [comment]
Hey 👋  @mraerino, there is no automatic notification. This feature will be available for use in our next nightly release which is a glimpse of what you can expect in the next v1.7.1 release which is set to go out next week. 

As far as notifications go we announce Packer releases to the Packer email distribution list. You can also subscribe to the GitHub RSS feeds so that you can see when new releases drop https://github.com/hashicorp/packer/releases.atom

Thanks again for making this change.

**Pull request merged**
